src/utils.jl

```diff
@@ -11,3 +11,14 @@ function _try_copy(data)
         deepcopy(data)
     end
 end
+
+function invert_dims(A::AbstractArray, dims)
+    ndims(A) == 1 && return dims
+    # TODO: support named dims
```

Comment:
Can you open an issue and link it in the code here?

Reply:
Let me know if this changes after Rory's input or not.

Reply:
No change - #20